source/python/neuropod/backends/torchscript/test/test_torchscript_dict_with_union_value_type_model.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (c) 2022 The Neuropod Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import unittest
+import torch
+
+import numpy as np
+
+from distutils.version import LooseVersion
+from typing import Dict, List, Union
+from testpath.tempdir import TemporaryDirectory
+
+from neuropod.packagers import create_torchscript_neuropod
+from neuropod.tests.utils import requires_frameworks
+
+
+class Model(torch.nn.Module):
+
+    def __init__(self, vocab: Dict[str, int]):
+        super().__init__()
+        self.vocab: Dict[str, int] = vocab
+
+    def preprocess(self, inputs: Union[List[str], torch.Tensor]) -> torch.Tensor:
+        # the following check is to help jit compiler know the type of inputs is List[str]
+        if isinstance(inputs, torch.Tensor):
+            raise ValueError("unsupported input type")
+        indexed_inputs: List[int] = []
+        for val in inputs:
+            indexed_inputs.append(self.vocab.get(val, 0))
+        return torch.tensor(indexed_inputs).reshape([len(indexed_inputs), 1])
+
+    def forward(self, inputs: Dict[str, Union[List[str], torch.Tensor]]) -> Dict[str, torch.Tensor]:
+        num_col = inputs['b']
+        # assertion is used to help compiler know that num_col is Tensor type
+        assert isinstance(num_col, torch.Tensor)
+        str_col = inputs['a']
+        return {'c': torch.cat([self.preprocess(str_col), num_col], dim=-1)}
+
+
+@unittest.skipIf(
+    LooseVersion(torch.__version__) < LooseVersion('1.10.0'),
+    "Torch supports UnionType after 1.10, skip the test if torch version < 1.10"
+)
+@requires_frameworks("torchscript")
+class TestTorchScriptDictWithUnionValueTypeModel(unittest.TestCase):
```

Comment:
This test seems to fail on GPU w/ Torch 1.10.

The error is 
```
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cpu and cuda:0! (when checking argument for argument tensors in method wrapper___cat)
```

The inputs to `cat` in the model above should all be on the same device.

See the logs here for more details: https://buildkite.com/uberopensource/neuropod/builds/171#ac931bc6-ead8-4bff-9ba0-c6e8664f6d52/368-2477

Reply:
@VivekPanyam could you grant me access to neuropod buildkite? Thanks!

Reply:
Working on it